packages/block-library/src/search/edit.native.js

```diff
@@ -154,112 +155,165 @@ export default function SearchEdit( {
+					isSelected={ isPlaceholderSelected }
+					className="wp-block-search__input"
+					style={ inputStyle }
+					numberOfLines={ 1 }
+					ellipsizeMode="tail" // currently only works on ios
+					label={ null }
+					value={ placeholder }
+					placeholder={
+						placeholder ? undefined : __( 'Optional placeholder…' )
+					}
+					onChange={ ( newVal ) =>
+						setAttributes( { placeholder: newVal } )
+					}
+					onFocus={ () => {
+						setIsPlaceholderSelected( true );
+						onFocus();
+					} }
+					onBlur={ () => setIsPlaceholderSelected( false ) }
+					placeholderTextColor={ placeholderStyle.color }
+				/>
+			</View>
 		);
 	};
 
+	// To achieve proper expanding and shrinking `RichText` on Android, there is a need to set
+	// a `placeholder` as an empty string when `RichText` is focused,
+	// because `AztecView` is calculating a `minWidth` based on placeholder text.
+	const buttonPlaceholderText =
+		isButtonSelected ||
+		( ! isButtonSelected && buttonText && buttonText !== '' )
+			? ''
+			: __( 'Add button text' );
+
 	const renderButton = () => {
 		return (
 			<View style={ styles.buttonContainer }>
-				{ buttonUseIcon && (
-					<Button
-						className="wp-block-search__button"
-						icon={ search }
-						onFocus={ onFocus }
-					/>
-				) }
+				{ buttonUseIcon && <Icon icon={ search } { ...styles.icon } /> }
 
 				{ ! buttonUseIcon && (
+					<View
+						accessible={ true }
+						accessibilityRole="none"
+						accessibilityHint={ __(
+							'Double tap to edit button text'
+						) }
+						accessibilityLabel={ `${ __(
+							'Search button. Current value is'
+						) } ${ buttonText }` }
+					>
+						<RichText
+							className="wp-block-search__button"
+							identifier="text"
+							tagName="p"
+							style={ styles.richTextButton }
+							placeholder={ buttonPlaceholderText }
+							value={ buttonText }
+							withoutInteractiveFormatting
+							onChange={ ( html ) =>
+								setAttributes( { buttonText: html } )
+							}
+							minWidth={ MIN_BUTTON_WIDTH }
+							textAlign="center"
+							isSelected={ isButtonSelected }
+							__unstableMobileNoFocusOnMount={ ! isSelected }
+							unstableOnFocus={ () => {
+								setIsButtonSelected( true );
+							} }
+							onBlur={ () => {
+								setIsButtonSelected( false );
+							} }
+							selectionColor={ styles.richTextButtonCursor.color }
+						/>
+					</View>
+				) }
+			</View>
+		);
+	};
+
+	return (
+		<View
+			{ ...blockProps }
+			style={ styles.searchBlockContainer }
+			importantForAccessibility={
+				isSelected ? 'yes' : 'no-hide-descendants'
+			}
+			accessibilityElementsHidden={ isSelected ? false : true }
+		>
+			{ isSelected && controls }
+
+			{ showLabel && (
+				<View
+					accessible={ true }
+					accessibilityRole="none"
+					accessibilityHint={ __( 'Double tap to edit label text' ) }
+					accessibilityLabel={ `${ __(
+						'Search block label. Current value is'
+					) } ${ label }` }
+				>
```

Comment:
I found on Android that when stepping through elements within a selected Search block, it would step through the wrapping element _and_ the input itself (i.e. 6 stops rather than 3). This did not occur on iOS. Did you experience this?

I'm not sure why difference between platforms would occur, but the multiple steps on Android feels a bit odd. Is the wrapping element necessary? I wonder if we could provide helpful custom labels/hints without the wrapping element?

Reply:
I agree, it's really annoying. Unfortunately customizing the `accessibilityLabel` and `accessibilityHint` only seems to work on container elements like `<View/>`. If I move those options directly to the element for editing, it no longer works.

Reply:
@AmandaRiu ah, thank you for providing that context. Relatedly, I found https://github.com/facebook/react-native/issues/26739 which appears to capture the root issue. If it is ever resolved, then we may be able to address the duplicative stops issue by removing the superfluous wrapping elements.